botocore/signers.py

```diff
@@ -410,33 +429,29 @@ def generate_presigned_post(self, request_dict, fields=None,
         for condition in conditions:
             policy['conditions'].append(condition)
 
-        # Obtain the appropriate signer.
-        query_prefix = '-presign-post'
-        signature_version = self._request_signer.signature_version
-        if not signature_version.endswith(query_prefix):
-            signature_version += query_prefix
-
-        kwargs = {'signing_name': self._request_signer.signing_name,
-                  'region_name': region_name,
-                  'signature_version': signature_version}
-
-        signature_type = signature_version.split('-', 1)[0]
-
-        try:
-            auth = self._request_signer.get_auth_instance(**kwargs)
-        except UnknownSignatureVersionError:
-            raise UnsupportedSignatureVersionError(
-                signature_version=signature_type)
+        signature_version = self._request_signer.choose_signer(
+            'PutObject', '-presign-post')
 
         # Store the policy and the fields in the request for signing
         request = create_request_object(request_dict)
         request.context['s3-presign-post-fields'] = fields
         request.context['s3-presign-post-policy'] = policy
 
-        auth.add_auth(request)
+        if signature_version is not botocore.UNSIGNED:
```

Comment:
This code seems like copy/paste from lines 220.  Why not fold this logic into the request signer itself, rather than exposing choose signer and forcing callers to all have similar logic?